zh/before-start_en.md

```diff
@@ -0,0 +1,62 @@
+# Before the beginning
+## The target audience for this document
+### The confused software developers
+* Developers who want to write code that can be easily read but don't know how to do that  
+* Developers who want to understand other people's code  
+### Software development beginner
+This is a document for software development beginners, who can be defined as having learned no more than two languages and being able to complete most of exercises in the course in terms of the language skills they have learned. From the general situation of college education, they could basicly complete the final course assignments arranged by teachers. We may not be able to say that they have done very well. It is also difficult to improve further.  
+
+For students who have graduated from college(plus or minus one or two years), it is difficult for them to get access to truly complex projects, and they will be incharge of the detailed parts even if they are involved in complex projects. They are eager to improve their software development ability quickly, but by accident, what they lack most is the opportunity to improve their ability.
+
+Therefore, this paper hopes to introduce a more reasonable method to help classmates to improve the ability of software development in a more scientific and reasonable way.
+### First-time users of open source software
+In the field of software development, it is almost impossible not to touch open source projects at all. How to improve the development efficiency with the help of open source projects in the daily development workand reduce the duplication of effort and how to benefit from open source rather than suffer from it is also an interesting topic. In a nutshell, it is risky to use the open source software without reading the source code(there is no different from using normal closed source software or closed source libraries). But if we can read and understand the source code, we can avoid this risk,and the better we understand it, the better we can use the software. Open source software offers this possibility to us.

```

Comment:
In the field of software development, it is almost impossible to ignore open source projects at all. How to improve the development efficiency with the help of open source projects in the daily development work and reduce the duplication of effort and how to benefit from open source rather than suffer from. it is also an interesting topic. In a nutshell, it is risky to use the open source software without reading the source code(there is no different from using normal closed source software or closed source libraries). But if we can read and understand the source code, we can avoid this risk,and the better we understand it, the better we can use the software. Open source software offers this possibility to us.